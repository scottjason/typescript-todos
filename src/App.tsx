import React, { useState } from 'react';
import { TodoList } from './TodoList';

const initialTodos: Todo[] = [
  {
    text: 'Walk the dog',
    isComplete: false,
  },
  {
    text: 'Build app',
    isComplete: true,
  },
];

const App: React.FC = () => {
  const [todos, setTodos] = useState(initialTodos);
  const onToggleTodo = (selectedTodo: Todo) => {
    const newTodos = todos.map(todo => {
      if (todo === selectedTodo) {
        todo.isComplete = !todo.isComplete;
      }
      return todo;
    });
    setTodos(newTodos);
  };
  return <TodoList todos={todos} onToggleTodo={onToggleTodo} />;
};

export default App;
