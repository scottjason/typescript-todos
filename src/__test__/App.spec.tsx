import userEvent from '@testing-library/user-event';
import { render, screen, RenderResult } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  let documentBody: RenderResult;
  beforeEach(() => {
    documentBody = render(<App />);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });

  it('should start with correct todos marked checked and unchecked', () => {
    expect(screen.getByText('Walk the dog')).toBeInTheDocument();
    const checkboxOne = screen.getByTestId('Walk the dog');
    expect(checkboxOne).not.toBeChecked();
    expect(screen.getByText('Build app')).toBeInTheDocument();
    const checkboxTwo = screen.getByTestId('Build app');
    expect(checkboxTwo).toBeChecked();
  });

  it('should be able to toggle between checked and unchecked', () => {
    const checkboxOne = screen.getByTestId('Walk the dog');
    expect(checkboxOne).not.toBeChecked();
    userEvent.click(checkboxOne);
    expect(checkboxOne).toBeChecked();
  });
});
