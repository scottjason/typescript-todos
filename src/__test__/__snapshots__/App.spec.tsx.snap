// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<body>
  <div>
    <ul>
      <li>
        <label>
          <input
            data-testid="Walk the dog"
            readonly=""
            type="checkbox"
          />
          Walk the dog
        </label>
      </li>
      <li>
        <label
          style="text-decoration: line-through;"
        >
          <input
            checked=""
            data-testid="Build app"
            readonly=""
            type="checkbox"
          />
          Build app
        </label>
      </li>
    </ul>
  </div>
</body>
`;
